realm/realm-library/src/androidTest/java/io/realm/RealmRecyclerAdapterTest.java

```diff
@@ -0,0 +1,296 @@
+        }
+    }
+
+    @Test
+    @UiThreadTest
+    public void removeItem_testRemoveFromAdapter() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        RealmRecyclerAdapter realmAdapter = new RealmRecyclerAdapter(context, resultList, automaticUpdate);
+        realmAdapter.removeItem(0);
+        assertEquals(TEST_DATA_SIZE - 1, realmAdapter.getItemCount());
+    }
+
+    @Test
+    @UiThreadTest
+    public void removeItem_testRemoveFromAdapterNoAutoUpdates() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        RealmRecyclerAdapter realmAdapter = new RealmRecyclerAdapter(context, resultList, false);
+        RecyclerView.AdapterDataObserver adapterObserver = mock(RecyclerView.AdapterDataObserver.class);
+        realmAdapter.registerAdapterDataObserver(adapterObserver);
+        realmAdapter.removeItem(0);
+
+        verify(adapterObserver, never()).onChanged();
+        verify(adapterObserver, never()).onItemRangeInserted(anyInt(), anyInt());
+        verify(adapterObserver, never()).onItemRangeMoved(anyInt(), anyInt(), anyInt());
+        verify(adapterObserver, never()).onItemRangeChanged(anyInt(), anyInt());
+        verify(adapterObserver, times(1)).onItemRangeRemoved(0, 1);
+    }
+
+    @Test
+    @UiThreadTest
+    public void updateData_realmResultInAdapter() {
+        RealmResults<AllTypes> resultList = testRealm.where(AllTypes.class).findAll();
+        resultList.sort(FIELD_STRING);
+        RealmRecyclerAdapter realmAdapter = new RealmRecyclerAdapter(context, resultList, false);
+        assertEquals(resultList.first().getColumnString(), realmAdapter.getData().first().getColumnString());
+        assertEquals(resultList.size(), realmAdapter.getData().size());
+
+        testRealm.beginTransaction();
+        AllTypes allTypes = testRealm.createObject(AllTypes.class);
+        allTypes.setColumnString("test data " + TEST_DATA_SIZE);
+        testRealm.commitTransaction();
+        assertEquals(resultList.last().getColumnString(), realmAdapter.getData().last().getColumnString());
+        assertEquals(resultList.size(), realmAdapter.getData().size());
+
+        RealmResults<AllTypes> emptyResultList = testRealm.where(AllTypes.class).equalTo(FIELD_STRING, "Not there").findAll();
+        realmAdapter.updateData(emptyResultList);
+        assertEquals(emptyResultList.size(), realmAdapter.getData().size());
+    }
+
+    @Test
+    @UiThreadTest
+    public void updateData_realmUnsupportedCollectionInAdapter() {
+        try {
+            RealmRecyclerAdapter realmAdapter = new RealmRecyclerAdapter(context, null, automaticUpdate);
+            realmAdapter.updateData(new UnsupportedCollection<AllTypes>());
+            fail("Should throw exception if there is unsupported collection");
+        } catch (IllegalArgumentException e) {
+        }
+    }
+
+    @Test
+    @UiThreadTest
+    public void testSortWithAdapter() {
```

Comment:
Please don't start JUnit4 tests names with `test` ... if you don't test a specific method, just use the smallest possible headline without the `test`-prefix. In this case you don't really test anything since after we merged the RealmCollection API, `sort` is no longer done in-place, but returns a sorted copy instead.